dopamine/utils/threading_utils.py

```diff
@@ -16,11 +16,23 @@
 
 The `local_attributes` class decorator defines a custom getter, setter and
 deleter for each specified attribute. These getter, setter and deleter actually
-wrap internal attributes that are thread specific.
+wrap internal attributes that are specific to the id of the thread they're
+accessed from.  We call those attributes "thread local".
 
-Each attribute has a callable default value that initializes the local value the
-first time they are called in a thread. To set these default values, use the
-`initialize_local_attributes` helper of this module.
+Each attribute has a callable default value that initializes a thread-local
+variable when it is not bound to a class at the time a caller tries to access
+it.  Typically, this is the first time in a thread that the variable is
+accessed.  Cases where this does not hold:
+
+  - If the thread itself is new but uses a pre-existing thread id, the variable
+    may have already existed earlier, and the default will not be called.
+
+  - If the variable var has at some point been unbound from the class cls (say
```

Comment:
isn't it `object`?

Reply:
This language has been swept up in my rewrite.